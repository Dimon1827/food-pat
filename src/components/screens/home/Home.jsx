import Service from "./service/Service";
import Stages from "./stages/Stages";


function Home() {
  return (
    <main class="main-container">
      <Service/>
      <Stages/>
    </main>
  );
}

export default Home;
