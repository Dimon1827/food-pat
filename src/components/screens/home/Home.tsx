import Example from './example/Example';
import Service from './service/Service';
import Stages from './stages/Stages';

function Home() {
  return (
    <main className="main-container" data-testid="home">
      <Service />
      <Stages />
      <Example />
    </main>
  );
}

export default Home;
