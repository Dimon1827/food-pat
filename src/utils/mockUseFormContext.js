import { useFormContext } from "react-hook-form";

export function mockUseFormContext(errors) {
  useFormContext.mockReturnValue({
    register: jest.fn(),
    formState: {
      errors: errors,
    },
  });
}


