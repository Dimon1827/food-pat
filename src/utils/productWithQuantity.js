const productWithQuantity = (food) => {
  return {
    ...food,
    isChecked: false,
    quantity: 1,
  };
};

export default productWithQuantity;
