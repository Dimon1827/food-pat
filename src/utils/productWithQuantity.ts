import { IProducts } from "redux/slice/productsSlice.types";

interface IproductWithQuantity extends Omit<IProducts, 'isChecked' | 'quantity'> {};

const productWithQuantity = (food: IproductWithQuantity) => {
  return {
    ...food,
    isChecked: false,
    quantity: 1,
  };
};

export default productWithQuantity;
