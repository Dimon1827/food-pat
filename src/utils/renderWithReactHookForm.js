import { render } from '@testing-library/react';
import { FormProvider, useForm } from 'react-hook-form';

export function renderWithReactHookForm(ui, { defaultValues = {} } = {}) {
  let reactHookFormMethods = {};

  const Wrapper = ({ children }) => {
    const methods = useForm({ defaultValues });
    return <FormProvider {...methods}>{children}</FormProvider>;
  };

  return {
    ...render(ui, { wrapper: Wrapper }),
  };
}