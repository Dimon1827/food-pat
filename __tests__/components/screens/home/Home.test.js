import Home from '@/components/screens/home/Home';
import { renderWithRouter } from '@/utils/renderWithRouter';

describe('Home', () => {
  it('Home renders correctly', () => {
    const { getByTestId } = renderWithRouter(<Home />);
    expect(getByTestId('home')).toMatchSnapshot();
  });
});
