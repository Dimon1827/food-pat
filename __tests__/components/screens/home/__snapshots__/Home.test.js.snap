// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Home renders correctly 1`] = `
<main
  class="main-container"
  data-testid="home"
>
  <section
    class="service"
    data-testid="service"
  >
    <h2
      class="visually-hidden"
    >
      Услуги
    </h2>
    <div
      class="container"
    >
      <ul
        class="service__list"
      >
        <li
          class="service__item"
        >
          <p
            class="service__description"
          >
            Лучшие органические продукты онлайн
          </p>
          <button
            class="service__link"
          >
            Купить
          </button>
        </li>
        <li
          class="service__item"
        >
          <p
            class="service__description"
          >
            Приготовьте свой завтрак Здоровый и легкий
          </p>
          <button
            class="service__link"
          >
            Купить
          </button>
        </li>
      </ul>
    </div>
  </section>
  <section
    class="stages"
    data-testid="stages"
  >
    <div
      class="container"
    >
      <h2
        class="stages__title title"
      >
        Преимущества
      </h2>
      <ul
        class="stages__list"
      >
        <li
          class="stages__item"
        >
          <h3
            class="stages__subtitle"
          >
            Качество
          </h3>
          <p
            class="stages__text"
          >
            Вкусно, полезно, доступно - здоровое питание для всех
          </p>
        </li>
        <li
          class="stages__item"
        >
          <h3
            class="stages__subtitle"
          >
            Многообразие
          </h3>
          <p
            class="stages__text"
          >
            Широкий выбор продуктов для активной и здоровой жизни
          </p>
        </li>
        <li
          class="stages__item"
        >
          <h3
            class="stages__subtitle"
          >
            Цена
          </h3>
          <p
            class="stages__text"
          >
            Качество по доступной цене! У нас выгодные цены на качественные продукты
          </p>
        </li>
        <li
          class="stages__item"
        >
          <h3
            class="stages__subtitle"
          >
            Подбор
          </h3>
          <p
            class="stages__text"
          >
            Поможем вам составить здоровый рацион и достичь целей
          </p>
        </li>
      </ul>
    </div>
  </section>
  <section
    class="example"
  >
    <div
      class="container example__container"
    >
      <div
        class="example__wrapper"
      >
        <h2
          class="example__title title"
        >
          Пример
        </h2>
        <p
          class="example__text"
        >
          “С продуктами Nest я легко составила здоровое меню! Качество и вкус на высшем уровне. За 4 месяца сбросила 18 кг. Чувствую себя бодрее и полна энергии! Рекомендую!”
        </p>
        <div
          class="example__stats"
        >
          <dl
            class="example__list"
          >
            <div
              class="example__item"
            >
              <dt
                class="example__description"
              >
                <span
                  class="example__description-text"
                >
                  снижение веса
                </span>
              </dt>
              <dd
                class="example__number"
              >
                18 КГ
              </dd>
            </div>
            <div
              class="example__item"
            >
              <dt
                class="example__description"
              >
                <span
                  class="example__description-text"
                >
                  затрачено времени
                </span>
              </dt>
              <dd
                class="example__number"
              >
                4 месяца
              </dd>
            </div>
          </dl>
          <span
            class="example__price"
          >
            Итог:
            <span
              class="example__price-number"
            >
              Довольна результатом
            </span>
          </span>
        </div>
      </div>
      <div
        class="slider"
        data-testid="slider"
      >
        <div
          class="example__slider-images slider__images"
          data-testid="slider-images"
        >
          <div
            class="slider__image-wrapper slider__image-wrapper--before"
            style="clip-path: inset(0 58% 0 0);"
          >
            <picture>
              <source
                height="737"
                media="(min-width:1280px)"
                srcset="./assets/images/woman-before-tablet.webp"
                type="image/webp"
                width="584"
              />
              <source
                height="737"
                media="(min-width:768px)"
                srcset="./assets/images/woman-before-tablet.webp"
                type="image/webp"
                width="584"
              />
              <source
                height="480"
                srcset="./assets/images/woman-before-mobile.webp"
                type="image/webp"
                width="312"
              />
              <source
                height="737"
                media="(min-width:1280px)"
                srcset="./assets/images/woman-before-tablet.png"
                type="image/png"
                width="584"
              />
              <source
                height="737"
                media="(min-width:768px)"
                srcset="./assets/images/woman-before-tablet.png"
                type="image/png"
                width="584"
              />
              <img
                alt="Женщина до"
                class="example__image"
                draggable="false"
                height="480"
                src="./assets/images/woman-before-mobile.png"
                width="312"
              />
            </picture>
          </div>
          <button
            class="example__button"
            style="left: calc(42% - 12px);"
            type="button"
          >
            <span
              class="example__button-toogle"
            />
            <span
              class="visually-hidden"
            >
              Кнопка для изменения пропорций
            </span>
          </button>
          <div
            class="slider__image-wrapper slider__image-wrapper--after"
          >
            <picture>
              <source
                height="737"
                media="(min-width:1280px)"
                srcset="./assets/images/woman-after-tablet.webp"
                type="image/webp"
                width="584"
              />
              <source
                height="737"
                media="(min-width:768px)"
                srcset="./assets/images/woman-after-tablet.webp"
                type="image/webp"
                width="584"
              />
              <source
                height="480"
                srcset="./assets/images/woman-after-mobile.webp"
                type="image/webp"
                width="312"
              />
              <source
                height="737"
                media="(min-width:1280px)"
                srcset="./assets/images/woman-after-tablet.png"
                type="image/png"
                width="584"
              />
              <source
                height="737"
                media="(min-width:768px)"
                srcset="./assets/images/woman-after-tablet.png"
                type="image/png"
                width="584"
              />
              <img
                alt="Женщина до"
                class="example__image"
                draggable="false"
                height="480"
                src="./assets/images/woman-after-mobile.png"
                width="312"
              />
            </picture>
          </div>
        </div>
      </div>
    </div>
  </section>
</main>
`;
